src/components/Button/Button.tsx

```diff
@@ -1,38 +1,56 @@
-import React, {useRef, useState, useCallback} from 'react';
+import React, {useCallback, useState} from 'react';
 import {CaretDownMinor} from '@shopify/polaris-icons';
 
 import {classNames, variationName} from '../../utilities/css';
-import {handleMouseUpByBlurring} from '../../utilities/focus';
+import {
+  handleMouseUpByBlurring,
+  MouseUpBlurHandler,
+} from '../../utilities/focus';
 import {useFeatures} from '../../utilities/features';
 import {useI18n} from '../../utilities/i18n';
 import {Icon} from '../Icon';
 import type {IconProps, ConnectedDisclosure} from '../../types';
 import {Spinner} from '../Spinner';
 import {Popover} from '../Popover';
 import {ActionList} from '../ActionList';
-import {UnstyledButton} from '../UnstyledButton';
+import {UnstyledButton, UnstyledButtonProps} from '../UnstyledButton';
 
 import styles from './Button.scss';
 
 type Size = 'slim' | 'medium' | 'large';
 type TextAlign = 'left' | 'right' | 'center';
 type IconSource = IconProps['source'];
 
-export interface ButtonProps {
+export interface ButtonProps
+  extends Pick<
```

Comment:
So I just tested this in the styleguide and it doesn't show any of these props in the docs with how this is setup. I think the better way to do this would be to have a `BaseButtonProps` interface that both `UnstyledButton` and `Button` extend for their props similar to how we do the [action types](https://github.com/Shopify/polaris-react/blob/master/src/types.ts#L86).

Reply:
Incredible 🙏  thank you so much for testing this! I've gone ahead and pushed a new commit with your suggestion 😺